proposals/0012-introduce-reactNativeMetadata.md

```diff
@@ -0,0 +1,447 @@
+
+So, for example, given the manifest:
+
+```json
+{
+  "reactNativeManifest": {
+    "windows": {
+      "capabilities": {
+        "hermes": {
+          "enabled": false
+        }
+      }
+    },
+  }
+}
+```
+
+It means that for all the platforms but Windows, React Native will use Hermes as JS engine (`hermes.enabled` default value is `true`). For Windows specifically, it would be disabled.
+
+As an extra note, please be careful to **follow the same semantics** as the React Native Core: in this way we can avoid confusion for our users.
+
+## Excluded use cases
+
+This section lists the use cases where we believe the `reactNativeManifest` section should **not** be used.
+
+### React Native version support
+
+For the time being, we're not planning to add a `version` section to the `reactNativeManifest` section for libraries. The rationale is that we believe that the `react-native` version should be specified in either the `peerDependencies` or the `engines` section of the `package.json` of the project. This follows the idiomatic way to specify the version of a dependency in a NodeJS project.
+
+### TurboModule/Fabric toggles
+
+At the time of writing, we prefer not to offer a dedicated section to toggle Fabric/TurboModule capability inside `reactNativeManifest`. The rationale is that we believe that the `newArchitecture` section is sufficient to expose the New Architecture to users.
+Selectively toggling Fabric/TurboModule is a more advanced feature. We believe we'll still be offering a more advanced way to enable or disable those pieces of the New Architecture infrastructure, but at the current state, our preference is not to expose such capability in the top level `reactNativeManifest` section.
+
+## Proposed Tooling
+
+This section lists the tooling we believe could be built on top of `reactNativeManifest` to improve the developer experience.
+
+### Build Logic support
+
+Build tools such as Gradle/CocoaPods or others should account for the `reactNativeManifest` section and use it to configure the build pipeline accordingly to this and future RFCs. Specifically they should follow this logic for a generic capability `foo`.
+
+1. If the capability `foo` is specificed in the `reactNativeManifest.capabilities` section, use the value specified in the `reactNativeManifest.capabilities.foo` section.
+   1. If the user is **also** specifying the capability `foo` in the build tool specific configuration (e.g. inside `gradle.properties` for Android) behave as follows:
+      1. If the two values are **compatible**, use them without notifying the user.
+      1. If the two values are **incompatible**, notify the user and use the value specified in the `reactNativeManifest.capabilities.foo` section will prevail and the value specified in the build tool specific configuration will be ignored.
```

Comment:
For some logic this is fine.  For build logic, the app has to be built either including or not including the new architecture files.  But for runtime logic I'm concerned about having the package.json values win.  How can you write experimentation code that flights A/B testing of enabling a capability if the value from package.json wins?

Reply:
I'm not sure I'm following. 
At runtime, you don't have access to the `package.json`. AFAIK, it is not shipped in the iOS bundle.
The way we expect it to work is that gradle/cocoapods reads the `package.json` and configures the build accordingly, eventually passing some flags to the build system.

It is actually already like that for some setitng like the NewArch enabled and hermes.
For the runtime flags, if those flags are controlled by equivalent build time flag (`#ifdefs`, to make it clear), you still can't change them at runtime. Otherwise, it is possible to write (and probably you already have) some code that fetch some configuration from a server and updates those flags.
In the case of A/B Tests, the package.json should specify the base configuration.

Reply:
gentle nudge for @acoates-ms and @tido64. 😊